storage_broker/src/bin/storage_broker.rs

```diff
@@ -369,8 +369,9 @@ impl BrokerService for Broker {
                     Err(RecvError::Lagged(skipped_msg)) => {
                         missed_msgs += skipped_msg;
                         if let Poll::Ready(_) = futures::poll!(Box::pin(warn_interval.tick())) {
-                            warn!("dropped {} messages, channel is full", missed_msgs);
-                            missed_msgs = 0;
+                            warn!("subscription id={}, key={:?}, addr={:?} dropped {} messages, channel is full",
+                                subscriber.id, subscriber.key, subscriber.remote_addr, missed_msgs);
+                            Err(Status::new(Code::Internal, "full channel"))?;
```

Comment:
This message seems wrong, since the subscriber is moving too slowly compared to the publisher.

I am also a bit unsure what does the `futures::poll!(Box::pin(warn_interval.tick()))` do, since it feels like it's used as a debounce to lagged events but will it not get reset if the channel goes back to normal..

Reply:
As such I worry that this might be hiding some loop which is causing new messages to be produced way too fast.

Regardless, originally I had the idea when suggesting the channel type that the messages are okay to lose, as in, the later message will always contain the superset of information from the older. As such, the broadcast channel buffer size could be small.